axiom/monitors.go

```diff
@@ -67,9 +82,31 @@ func (c *Operator) UnmarshalJSON(b []byte) (err error) {
 	return err
 }
 
+// MarshalJSON implements [json.Marshaler]. It is in place to marshal the
+// Type to its string representation because that's what the server
+// expects.
+func (c Type) MarshalJSON() ([]byte, error) {
+	return json.Marshal(c.String())
+}
+
+// UnmarshalJSON implements [json.Unmarshaler]. It is in place to unmarshal the
+// Type from the string representation the server returns.
+func (c *Type) UnmarshalJSON(b []byte) (err error) {
+	var s string
+	if err = json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+
+	*c = typeFromString(s)
+
+	return nil
+}
```

Comment:
Please group these methods closer to the type declaration. E.g. every type, its constants and the methods are a logical block of code, e.g. for `Operator`:

```go
// Operator represents a comparison operation for a monitor. A [Monitor] acts on
// the result of comparing a query result with a threshold.
type Operator uint8

// All available [Monitor] comparison [Operator]s.
const (
	emptyOperator Operator = iota //

	Below        // Below
	BelowOrEqual // BelowOrEqual
	Above        // Above
	AboveOrEqual // AboveOrEqual
)

func operatorFromString(s string) (c Operator, err error) {
	switch s {
	case emptyOperator.String():
		c = emptyOperator
	case Below.String():
		c = Below
	case BelowOrEqual.String():
		c = BelowOrEqual
	case Above.String():
		c = Above
	case AboveOrEqual.String():
		c = AboveOrEqual
	default:
		err = fmt.Errorf("unknown operator %q", s)
	}

	return c, err
}

// MarshalJSON implements [json.Marshaler]. It is in place to marshal the
// Operator to its string representation because that's what the server
// expects.
func (c Operator) MarshalJSON() ([]byte, error) {
	return json.Marshal(c.String())
}

// UnmarshalJSON implements [json.Unmarshaler]. It is in place to unmarshal the
// Operator from the string representation the server returns.
func (c *Operator) UnmarshalJSON(b []byte) (err error) {
	var s string
	if err = json.Unmarshal(b, &s); err != nil {
		return err
	}

	*c, err = operatorFromString(s)

	return err
}
```